apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/PromptIntegrations/utils.ts

```diff
@@ -1,202 +0,0 @@
-  )
-  if (integrationTools.includes(`${integrationName}/*`)) return config
-  if (integrationTools.includes(`${integrationName}/${toolName}`)) return config
-
-  if (toolName === '*') {
-    const otherIntegrationTools = integrationTools.filter(
-      (integrationId) => integrationId.split('/')[0] !== integrationName,
-    )
-    config.tools = [
-      ...clientTools,
-      ...otherIntegrationTools,
-      `${integrationName}/*`,
-    ]
-    return config
-  }
-
-  config.tools = [
-    ...clientTools,
-    ...integrationTools,
-    `${integrationName}/${toolName}`,
-  ]
-  return config
-}
-
-function removeIntegrationFromConfig({
-  config,
-  integrationName: removedIntegrationName,
-  toolName: removedToolName,
-  integrationToolNames,
-}: {
-  config: LatitudePromptConfig
-  integrationName: string
-  toolName: string
-  integrationToolNames: string[]
-}) {
-  if (config.tools) config = normalizeIntegrations(config)
-  const tools = (config.tools as ToolsItem[]) ?? []
-
-  if (removedToolName === '*') {
-    config.tools = tools.filter((tool: ToolsItem) => {
-      if (typeof tool !== 'string') return true
-      return !tool.startsWith(`${removedIntegrationName}/`)
-    })
-    return config
-  }
-
-  const integrationToolsToAdd = tools.includes(`${removedIntegrationName}/*`)
-    ? integrationToolNames.filter((tn) => tn !== removedToolName)
-    : []
-
-  config.tools = tools
-    .filter((tool: ToolsItem) => {
-      if (typeof tool !== 'string') return true
-      const [integrationName, toolName] = tool.split('/')
-      if (integrationName !== removedIntegrationName) return true
-      if (toolName === '*') return false
-      if (integrationToolsToAdd.includes(toolName ?? '*')) return false
-      return toolName !== removedToolName
-    })
-    .concat(
-      integrationToolsToAdd.map((tn) => `${removedIntegrationName}/${tn}`),
-    )
-
-  return config
-}
-
-export const useActiveIntegrations = ({
-  config: promptConfig,
-  setConfig: setPromptConfig,
-  integrations,
-}: {
-  config: Record<string, unknown>
-  setConfig: (config: Record<string, unknown>) => void
-  integrations: IntegrationDto[]
-  disabled?: boolean
-}) => {
-  const [localConfig, setLocalConfig] = useState<LatitudePromptConfig>(
-    promptConfig as LatitudePromptConfig,
-  )
-  const activeIntegrations = useMemo(
-    () => readActiveIntegrations({ config: localConfig, integrations }),
-    [localConfig, integrations],
-  )
-  const debouncedSetPromptConfig = useDebouncedCallback(
-    (config: Record<string, unknown>) => {
-      setPromptConfig(config)
-    },
-    1000,
-    { trailing: true },
-  )
-
-  useEffect(() => {
-    setLocalConfig(promptConfig as LatitudePromptConfig)
-  }, [promptConfig])
-
-  const setConfig = useCallback(
-    (config: Record<string, unknown>) => {
-      setLocalConfig(config as LatitudePromptConfig)
-      debouncedSetPromptConfig(config)
-    },
-    [debouncedSetPromptConfig, setLocalConfig],
-  )
```

Comment:
The problem was that we were using a `useEffect` and also this debounce. I refactored to use `useEvents`, which allows me to receive input from outside. However, the internal state that this hook (new hook `useActiveIntegrations`) outputs is only updated when an action occurs.